scipy/io/matlab/mio5.py

```diff
@@ -5,17 +5,8 @@
 from scipy._lib.deprecation import _sub_module_deprecation
 
 __all__ = [  # noqa: F822
-    'mclass_info', 'mxCHAR_CLASS', 'mxSPARSE_CLASS',
-    'BytesIO', 'native_code',
-    'swapped_code', 'MatFileReader', 'docfiller', 'matdims',
-    'read_dtype', 'arr_to_chars', 'arr_dtype_number', 'MatWriteError',
-    'MatReadError', 'MatReadWarning', 'VarReader5', 'MatlabObject',
-    'MatlabFunction', 'MDTYPES', 'NP_TO_MTYPES', 'NP_TO_MXTYPES',
-    'miCOMPRESSED', 'miMATRIX', 'miINT8', 'miUTF8', 'miUINT32',
-    'mxCELL_CLASS', 'mxSTRUCT_CLASS', 'mxOBJECT_CLASS', 'mxDOUBLE_CLASS',
-    'mat_struct', 'ZlibInputStream', 'MatFile5Reader', 'varmats_from_mat',
```

Comment:
Can we leave `'varmats_from_mat'` in and skip it in the test?  #19223 is unresolved so think it would be safer to sit on it and get everything else merged

Reply:
I looked at doing that, but I don't want to add special handling in `test_public_api.py`. We should be dogfooding here - I've put it in the `scipy.io.matlab` namespace (which solves the warning confusion portion of #19223). Given that this was an explicit recommendation in the warning, I think we cannot really consider the function to be really private anymore anyway...

If we don't like the name (I certainly think it's... suboptimal), we should come up with a better alternative before the next release.

Reply:
Ok that's fine with me too. Two final points then I think this is good to go, can you update the version here
https://github.com/scipy/scipy/blob/12a231cea3cfa3e46102da7bf62ac5e138c3e8d8/scipy/_lib/deprecation.py#L61
so that we can remove `varmats_from_mat` from `scipy.io.matlab.mio5` in the future and could you post to the development forum as we are now adding to the public api

Reply:
> as we are now adding to the public api

Since our warnings have been pointing to them, the function must be considered public already. I really dislike its name, but that's what it is.

We can immediately redeprecate it if we want, as long as we provide the functionality somewhere else. In any case, I don't think we'll fix this in time for 1.14, so I'm tending back to your suggestion to skip it in the test and unblock the rest.

Reply:
Either is fine with me (I am not familiar enough with the `io` submodule to put forward a better name!)